src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+type TLocalStorageValue = string | boolean | number;
+
+interface IUseLocalStorageProps<T extends TLocalStorageValue> {
+  localStorageKey: string;
+  initialValue: T;
+  localStorageValueParser?: (localStorageValue: string) => T;
+}
+
+export const useLocalStorage = <T extends TLocalStorageValue>({
```

Comment:
Optionally this could be made more general to include `sessionStorage`, etc. -> `useStorage` accepting additional parameter defining which storage should be used.

An alternative solution might involve creating separate hooks like `useSessionStorage`, etc. but this approach could lead to unnecessary code duplication.